### PR TITLE
BFD-2420: Upgrade Out of Date Dependencies

### PR DESCRIPTION
**Pull request by JeremyIR**
**JIRA Ticket:**
[BFD-2420](https://jira.cms.gov/browse/BFD-2420)

**User Story or Bug Summary:**

This PR addresses several library/dependency updates. 

SNYK and Performance Test Results:
- Performance regression tests were conducted on both this branch with upgraded dependency versions, and the master branch on TEST server. The number of requests differed by 2.07%, and the average number of requests/second differed by 1.82%.

<details>
<summary>
TEST Server Master Branch

</summary>

```
Type     Name                                                                          # reqs      # fails \|    Avg     Min     Max    Med \|   req/s  failures/s
--
--------\|----------------------------------------------------------------------------\|-------\|-------------\|-------\|-------\|-------\|-------\|--------\|-----------
GET      /v2/fhir/Coverage search by id                                                   208     0(0.00%) \|     43      10     773     16 \|    6.64        0.00
GET      /v2/fhir/Coverage search by id / count=10                                        192     0(0.00%) \|     35      10     687     15 \|    6.13        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id                                       181     0(0.00%) \|    422     156    1956    320 \|    5.78        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id / count=10                            201     0(0.00%) \|    293     155    1163    230 \|    6.42        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id / includeTaxNumbers                   196     0(0.00%) \|    376     155    2085    310 \|    6.26        0.00
GET      /v2/fhir/Patient search by coverage contract (all pages)                         187     0(0.00%) \|     51      14    1291     21 \|    5.97        0.00
GET      /v2/fhir/Patient search by hashed mbi / includeIdentifiers = mbi                 198     0(0.00%) \|     27       6     625     10 \|    6.32        0.00
GET      /v2/fhir/Patient search by id                                                    200     0(0.00%) \|     25       5     582      8 \|    6.39        0.00
GET      /v2/fhir/Patient search by id / _IncludeIdentifiers=mbi                          202     0(0.00%) \|     29       5     804      8 \|    6.45        0.00
--------\|----------------------------------------------------------------------------\|-------\|-------------\|-------\|-------\|-------\|-------\|--------\|-----------
Aggregated                                                                      1765     0(0.00%) \|    142       5    2085     23 \|   56.37        0.00
Response time percentiles (approximated)
Type     Name                                                                                  50%    66%    75%    80%    90%    95%    98%    99%  99.9% 99.99%   100% # reqs
--------\|--------------------------------------------------------------------------------\|--------\|------\|------\|------\|------\|------\|------\|------\|------\|------\|------\|------
GET      /v2/fhir/Coverage search by id                                                         16     27     34     39    130    160    170    330    770    770    770    208
GET      /v2/fhir/Coverage search by id / count=10                                              15     23     27     29     61    140    240    340    690    690    690    192
GET      /v2/fhir/ExplanationOfBenefit search by id                                            320    390    480    530    660    930   1700   1900   2000   2000   2000    181
GET      /v2/fhir/ExplanationOfBenefit search by id / count=10                                 230    290    320    370    480    640    900    970   1200   1200   1200    201
GET      /v2/fhir/ExplanationOfBenefit search by id / includeTaxNumbers                        310    370    410    440    640    800   1200   1700   2100   2100   2100    196
GET      /v2/fhir/Patient search by coverage contract (all pages)                               21     24     29     35     97    160    450    760   1300   1300   1300    187
GET      /v2/fhir/Patient search by hashed mbi / includeIdentifiers = mbi                       10     12     15     19     68    140    210    300    620    620    620    198
GET      /v2/fhir/Patient search by id                                                           8      9     11     14     71    130    210    280    580    580    580    200
GET      /v2/fhir/Patient search by id / _IncludeIdentifiers=mbi                                 8     11     13     16     39    130    240    310    800    800    800    202
--------\|--------------------------------------------------------------------------------\|--------\|------\|------\|------\|------\|------\|------\|------\|------\|------\|------\|------
Aggregated                                                                             23    170    220    260    380    520    760    940   2000   2100   2100   1765

```

</details>

<details>
<summary>
TEST Server UPGRADES:

</summary>

```
Type     Name                                                                          # reqs      # fails \|    Avg     Min     Max    Med \|   req/s  failures/s
--
--------\|----------------------------------------------------------------------------\|-------\|-------------\|-------\|-------\|-------\|-------\|--------\|-----------
GET      /v2/fhir/Coverage search by id                                                   168     0(0.00%) \|     41      10     995     14 \|    5.35        0.00
GET      /v2/fhir/Coverage search by id / count=10                                        195     0(0.00%) \|     42      11     784     15 \|    6.21        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id                                       216     0(0.00%) \|    375     147    2131    290 \|    6.88        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id / count=10                            202     0(0.00%) \|    293     143    1594    220 \|    6.44        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id / includeTaxNumbers                   188     0(0.00%) \|    413     144    2453    290 \|    5.99        0.00
GET      /v2/fhir/Patient search by coverage contract (all pages)                         204     0(0.00%) \|     57      15    1212     24 \|    6.50        0.00
GET      /v2/fhir/Patient search by hashed mbi / includeIdentifiers = mbi                 209     0(0.00%) \|     32       6     862     11 \|    6.66        0.00
GET      /v2/fhir/Patient search by id                                                    190     0(0.00%) \|     21       6     632      9 \|    6.05        0.00
GET      /v2/fhir/Patient search by id / _IncludeIdentifiers=mbi                          230     0(0.00%) \|     28       5     620      9 \|    7.33        0.00
--------\|----------------------------------------------------------------------------\|-------\|-------------\|-------\|-------\|-------\|-------\|--------\|-----------
Aggregated                                                                      1802     0(0.00%) \|    145       5    2453     23 \|   57.41        0.00
Response time percentiles (approximated)
Type     Name                                                                                  50%    66%    75%    80%    90%    95%    98%    99%  99.9% 99.99%   100% # reqs
--------\|--------------------------------------------------------------------------------\|--------\|------\|------\|------\|------\|------\|------\|------\|------\|------\|------\|------
GET      /v2/fhir/Coverage search by id                                                         14     18     21     22     85    130    240    960   1000   1000   1000    168
GET      /v2/fhir/Coverage search by id / count=10                                              15     17     24     30     74    160    620    630    780    780    780    195
GET      /v2/fhir/ExplanationOfBenefit search by id                                            290    340    410    420    590    870   1600   1700   2100   2100   2100    216
GET      /v2/fhir/ExplanationOfBenefit search by id / count=10                                 220    260    300    330    460    730   1000   1400   1600   1600   1600    202
GET      /v2/fhir/ExplanationOfBenefit search by id / includeTaxNumbers                        290    370    420    490    800   1200   1700   1800   2500   2500   2500    188
GET      /v2/fhir/Patient search by coverage contract (all pages)                               24     28     34     48    140    170    250    560   1200   1200   1200    204
GET      /v2/fhir/Patient search by hashed mbi / includeIdentifiers = mbi                       11     13     19     23     74    150    210    330    860    860    860    209
GET      /v2/fhir/Patient search by id                                                           9     10     11     14     24     62    150    560    630    630    630    190
GET      /v2/fhir/Patient search by id / _IncludeIdentifiers=mbi                                 9     10     12     18     40    130    280    450    620    620    620    230
--------\|--------------------------------------------------------------------------------\|--------\|------\|------\|------\|------\|------\|------\|------\|------\|------\|------\|------
Aggregated                                                                             23    160    210    250    370    550    860   1200   2100   2500   2500   1802
```

</details>

### What Does This PR Do?

Change library versions to more secure.

### What Should Reviewers Watch For?

Does the code follow style guidelines?
Have all the libraries in question been addressed and contribute to lower vulnerability when running snyk?
If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-model/bfd-model-rif-samples/src/test/java/gov/cms/bfd/model/rif/samples/SampleDataColumnsTest.java [hunk 1]
```diff
@@ -103,8 +102,8 @@ private void verifyColumns(StaticRifResourceGroup sampleGroup) {
                   toHeaderFormat(columnsInEnum, c -> c.name())));
         }
       }
-    } catch (IOException e) {
-      throw new UncheckedIOException(e);
+    } catch (UncheckedIOException e) {
+      throw e;
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rif-samples/src/test/java/gov/cms/bfd/model/rif/samples/SampleDataColumnsTest.java:106]: Not exactly sure what the point of this statement is anyway; if it didn't exist it'd do the same thing.

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/pom.xml [hunk 2]
```diff
@@ -63,7 +63,7 @@
         <dependency>
             <groupId>org.freemarker</groupId>
             <artifactId>freemarker</artifactId>
-            <version>2.3.31</version>
+            <version>2.3.32</version>
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/pom.xml:66]: IMO we should pull all these misc version numbers into the parent pom so our versions are all together.

File: apps/pom.xml [hunk 3]
```diff
@@ -75,54 +75,54 @@
...
-		<jackson-dataformat-yaml.version>2.14.1</jackson-dataformat-yaml.version>
-		<jackson-datatype-jsr310.version>2.14.1</jackson-datatype-jsr310.version>
+		<jackson-databind.version>2.14.2</jackson-databind.version>
+		<jackson-datatype-jdk8.version>2.14.2</jackson-datatype-jdk8.version>
+		<jackson-dataformat-yaml.version>2.14.2</jackson-dataformat-yaml.version>
+		<jackson-datatype-jsr310.version>2.14.2</jackson-datatype-jsr310.version>
 
-		<mockito.version>4.11.0</mockito.version>
-		<postgresql.version>42.5.1</postgresql.version>
+		<mockito.version>5.1.1</mockito.version>
```
Comment (lsmitchell) [inline apps/pom.xml:99]: Big deal; mockito 5 allows us to mock static and final classes without a bunch of workaround now!

File: apps/bfd-model/bfd-model-rif-samples/src/test/java/gov/cms/bfd/model/rif/samples/SampleDataColumnsTest.java (hunk 1 above)
Comment (JeremyIR) [reply apps/bfd-model/bfd-model-rif-samples/src/test/java/gov/cms/bfd/model/rif/samples/SampleDataColumnsTest.java:106]: This is why I made this change. One of the dependencies I upgraded caused it to fail. Screenshot 2023-02-20 at 2 09 37 PM

Comment (lsmitchell) [reply apps/bfd-model/bfd-model-rif-samples/src/test/java/gov/cms/bfd/model/rif/samples/SampleDataColumnsTest.java:106]: I'm suggesting just take that catch out completely, it did (and continues to do) nothing.

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/pom.xml (hunk 2 above)
Comment (JeremyIR) [reply apps/bfd-pipeline/bfd-pipeline-rda-bridge/pom.xml]: Fixed

File: apps/bfd-model/bfd-model-rif-samples/src/test/java/gov/cms/bfd/model/rif/samples/SampleDataColumnsTest.java (hunk 1 above)
Comment (JeremyIR) [reply apps/bfd-model/bfd-model-rif-samples/src/test/java/gov/cms/bfd/model/rif/samples/SampleDataColumnsTest.java]: Removed

**Review by lsmitchell — APPROVED** [review]

**Comment by lsmitchell** [comment]
Approved for the code changes, but check on that synk thing.

**Review by brianburton — APPROVED** [review]
Changes look fine to me.

**Pull request merged**